### PR TITLE
Remove or fix azure* environment variables in live tests

### PR DESCRIPTION
**Pull request by benbp**
Remove or fix azure* environment variables in live tests

### CODE REVIEW

File: sdk/textanalytics/tests.yml [hunk 1]
```diff
@@ -12,15 +12,8 @@ extends:
             - $(sub-config-text-analytics-azure-cloud-test-resources)
           MatrixReplace:
             - TestSamples=.*/true
-        UsGov:
```
Comment (weshaggard) [inline sdk/textanalytics/tests.yml:15]: Did you intend to remove these configs?

Comment (benbp) [reply sdk/textanalytics/tests.yml:15]: I'm forgetting, aren't they not necessary anymore?

Comment (weshaggard) [reply sdk/textanalytics/tests.yml:15]: If you override one config you need to list all them in your custom config list.

**Review by weshaggard — APPROVED** [review]

File: sdk/keyvault/tests.yml [hunk 2]
```diff
@@ -26,5 +26,5 @@ extends:
       EnvVars:
         AZURE_TEST_RUN_LIVE: true
         AZURE_SKIP_LIVE_RECORDING: 'True'
-        AZURE_SUBSCRIPTION_ID: $(azure-subscription-id)
+        AZURE_SUBSCRIPTION_ID: $(KEYVAULT_SUBSCRIPTION_ID)
```
Comment (weshaggard) [inline sdk/keyvault/tests.yml:29]: Same question here isn't this automatically set?

File: sdk/cognitivelanguage/tests.yml [hunk 3]
```diff
@@ -18,10 +18,7 @@ extends:
         AZURE_CONVERSATIONS_DEPLOYMENT_NAME: $(clu-project-deployment)
         AZURE_CONVERSATIONS_WORKFLOW_PROJECT_NAME: $(clu-workflow-project)
         AZURE_CONVERSATIONS_WORKFLOW_DEPLOYMENT_NAME: $(clu-workflow-project-deployment)
-        AZURE_CLIENT_ID: $(aad-azure-sdk-test-client-id)
-        AZURE_CLIENT_SECRET: $(aad-azure-sdk-test-client-secret)
-        AZURE_SUBSCRIPTION_ID: $(azure-subscription-id)
-        AZURE_TENANT_ID: $(aad-azure-sdk-test-tenant-id)
+        AZURE_SUBSCRIPTION_ID: $(COGNITIVELANGUAGE_SUBSCRIPTION_ID)
```
Comment (weshaggard) [inline sdk/cognitivelanguage/tests.yml:21]: Isn't this automatically set?

File: sdk/storage/tests.yml [hunk 4]
```diff
@@ -31,4 +31,4 @@ extends:
       EnvVars:
         AZURE_SKIP_LIVE_RECORDING: 'True'
         AZURE_TEST_RUN_LIVE: 'true'
-        AZURE_SUBSCRIPTION_ID: $(azure-subscription-id)
+        AZURE_SUBSCRIPTION_ID: $(STORAGE_SUBSCRIPTION_ID)
```
[the remark this answers is not in the document]
Comment (benbp) [reply sdk/storage/tests.yml:34]: The `AZURE` prefix variables are not set automatically. I removed it in places where the test code has no reference to it (i.e. where it was only used for EnvironmentCredential in the past).

**Comment by benbp** [comment]
test

File: eng/pipelines/templates/steps/build-test.yml [hunk 5]
```diff
@@ -149,9 +149,11 @@ steps:
       condition: and(succeeded(), eq(variables['TestSamples'], 'true'))
       env:
         SYSTEM_ACCESSTOKEN: $(System.AccessToken)
+        # Enable samples tests that use DefaultAzureCredential to load the federated pipeline credential
+        AZURE_POD_IDENTITY_AUTHORITY_HOST: 'https://FakeAuthorityHost'
         ${{ insert }}: ${{ parameters.EnvVars }}
       inputs:
-        azureSubscription: ${{ parameters.ServiceConnection }}
+        azureSubscription: azure-sdk-tests
```
[the remark this answers is not in the document]
Comment (benbp) [reply eng/pipelines/templates/steps/build-test.yml:156]: These are for the samples tests which in some cases still rely on static resources in the old subscription, same deal as with Java where we decided not to migrate for now

[the remark this answers is not in the document]
Comment (weshaggard) [reply eng/pipelines/templates/steps/build-test.yml:156]: I thought samples had their own arm templates if we were running them live. We should definitely be careful using static resources for testing samples as these are meant to be copy/pasted by customers.

**Pull request merged**